app/api/tickets.py

```diff
@@ -89,6 +90,14 @@ def query(self, view_kwargs):
             # access_code - ticket :: many-to-many relationship
             query_ = Ticket.query.filter(Ticket.access_codes.any(id=access_code.id))
 
+        if view_kwargs.get('discount_code_id'):
+            discount_code = safe_query(
+                self, DiscountCode, 'id', view_kwargs['discount_code_id'],
+                'discount_code_id')
+            # discount_code - ticket :: one-to-many relationship
```

Comment:
many to many

Reply:
I guess I was told earlier that only one discount code can be applied to a given ticket, although different tickets can have the same discount code.

Reply:
Check the legacy server, it allows to create two discount codes on same ticket.

Reply:
From the chat:
> Let's consider an example:
Let's say you're the event organizer and you only have one ticket in the event. You want to provide special discount of 10% to your friends and 5% to your relatives. How will you provide it if you don't allow tickets to have multiple associated discount codes ? Makes sense ?